#! /bin/sh /usr/share/dpatch/dpatch-run
## 00_0175-fix-s390-qatomic.dpatch by Dirk Mueller <mueller@kde.org>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Fix s390(x) build.

@DPATCH@
diff -urNad qt4-x11-4.3.0~rc1~/src/corelib/arch/qatomic_s390.h qt4-x11-4.3.0~rc1/src/corelib/arch/qatomic_s390.h
--- qt4-x11-4.3.0~rc1~/src/corelib/arch/qatomic_s390.h	2007-05-06 17:54:22.000000000 +0200
+++ qt4-x11-4.3.0~rc1/src/corelib/arch/qatomic_s390.h	2007-05-17 19:13:18.000000000 +0200
@@ -182,10 +182,20 @@
 #endif
 }
 
-#error "fetch-and-add not implemented"
-// int q_atomic_fetch_and_add_int(volatile int *ptr, int value);
-// int q_atomic_fetch_and_add_acquire_int(volatile int *ptr, int value);
-// int q_atomic_fetch_and_add_release_int(volatile int *ptr, int value);
+inline int q_atomic_fetch_and_add_int(volatile int *ptr, int value)
+{
+    return __sync_fetch_and_add(ptr, value);
+}
+
+inline int q_atomic_fetch_and_add_acquire_int(volatile int *ptr, int value)
+{
+    return __sync_fetch_and_add(ptr, value);
+}
+
+inline int q_atomic_fetch_and_add_release_int(volatile int *ptr, int value)
+{
+    return __sync_fetch_and_add(ptr, value);
+}
 
 QT_END_HEADER
 
